File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* temp=head;
        int num=1;
        while(temp->next!=nullptr){
            num++;
            temp=temp->next;
        }
        temp=head;
        if(num==1){
            delete head;
            return nullptr;
        }
        if(num==n){
            head=temp->next;
            return head;
        }
        for(int i=0;i<num-n-1;i++)
            temp=temp->next;
        ListNode *rem=temp->next;
        temp->next=temp->next->next;
        delete rem;
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        prev = None
        curr = head
        while curr:
            temp = curr.next
            curr.next = prev
            prev = curr
            curr = temp
        rhead = prev

        dummy = ListNode(0, rhead)
        curr = dummy
        for _ in range(n - 1):
            if curr.next:
                curr = curr.next
        if curr.next:
            curr.next = curr.next.next

        prev = None
        curr = dummy.next
        while curr:
            temp = curr.next
            curr.next = prev
            prev = curr
            curr = temp

        return prev



        